src/graph/type_indexededgelist.c

```diff
@@ -483,8 +554,30 @@ igraph_error_t igraph_delete_edges(igraph_t *graph, igraph_es_t edges) {
     IGRAPH_FINALLY_CLEAN(1);
 
     /* Create start vectors, no memory is needed for this */
-    igraph_i_create_start(&graph->os, &graph->from, &graph->oi, no_of_nodes);
-    igraph_i_create_start(&graph->is, &graph->to,   &graph->ii, no_of_nodes);
+    igraph_i_create_start_vectors(&graph->os, &graph->from, &graph->oi, no_of_nodes);
+    igraph_i_create_start_vectors(&graph->is, &graph->to,   &graph->ii, no_of_nodes);
+
+    /* modification successful, clear the cached properties of the graph.
+     *
+     * Deleting one or more edges cannot make a directed acyclic graph cyclic,
+     * or an undirected forest into a cyclic graph, so we keep those flags if
+     * they are cached as true
+     *
+     * Also, if the graph had no loop edges before the deletion, it will have
+     * no loop edges after the deletion either. The same applies to reciprocal
+     * edges or multiple edges as well.
+     */
```

Comment:
I think we can also have a `keep_when_false` for `is_strongly_connected` and `is_weakly_connected`.

Is this correct or am I missing something?

Reply:
Hmmm, right, I was thinking like removing all the edges of all connected components but one, but then we still have the vertices there so we are okay.

Reply:
Yes, removing vertices is different, and we can't keep any cached connectedness properties there. But removing edges is simpler: it can never make a disconnected graph be connected.